tests/Unit/NumericalAlgorithms/LinearOperators/Test_IndefiniteIntegral.cpp

```diff
@@ -0,0 +1,113 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/StripeIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/LinearOperators/IndefiniteIntegral.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+template <size_t Dim, typename T>
+DataVector integrand(const std::array<size_t, Dim>& exponent,
+                     const T& x) noexcept {
+  DataVector integrand(get<0>(x).size(), 1.0);
+  for (size_t d = 0; d < Dim; ++d) {
+    integrand *= pow(x.get(d), gsl::at(exponent, d));
+  }
+  return integrand;
+}
+
+template <size_t Dim, typename T>
+DataVector integral(const std::array<size_t, Dim>& exponent, const T& x,
+                    const Mesh<Dim> mesh,
+                    const size_t only_this_dim = 100) noexcept {
+  DataVector integral(get<0>(x).size(), 1.0);
+  for (size_t d = 0; d < Dim; ++d) {
+    if (only_this_dim == 100 or d == only_this_dim) {
```

Comment:
makes me sad for us not having `std::optional` support.

Reply:
There is `boost::optional`, but since we have only integration in a specific dim going forward, I'm going to remove the default :)